### PR TITLE
Fix openshift compat for kubefed join and e2e

### PR DESCRIPTION
**Pull request by marun**
Openshift >= 1.7 needs the included changes to support ``kubefed join`` (since openshift creates an extra secret per service account) and e2e (since secrets no longer contain ``kubeconfig`` as of 1.7).

/assign irfanurrehman shashidharatd 

cc: @onyiny-ang @font

### CODE REVIEW

File: test/e2e/framework/cluster.go [hunk 1]
```diff
@@ -209,28 +209,39 @@ func clusterConfigFromSecrets(f *Framework) []*clusterConfig {
...
+		var err error
+		// Host address will be populated on client creation, so use a temp address here
+		clusterConfig, err = clientcmd.BuildConfigFromKubeconfigGetter("TEMP ADDRESS", func() (*clientcmdapi.Config, error) {
+			data, ok := secret.Data[util.KubeconfigSecretDataKey]
+			if !ok {
+				return nil, fmt.Errorf("secret does not have data with key: %s", util.KubeconfigSecretDataKey)
+			}
+			return clientcmd.Load(data)
+		})
+		framework.ExpectNoError(err, fmt.Sprintf("Error building loading config secret \"%s/%s\" for cluster %q: %+v", namespace, secretName, clusterName, err))
```
Comment (font) [inline test/e2e/framework/cluster.go]: s/loading //

File: pkg/kubefed/join.go [hunk 2]
```diff
@@ -649,28 +649,25 @@ func populateSecretInHostCluster(clusterClientset, hostClientset internalclients
...
-			return false, nil
+		for _, objReference := range sa.Secrets {
+			secretName := objReference.Name
+			var err error
+			secret, err = clusterClientset.Core().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+			if err != nil {
+				return false, nil
+			}
+			if secret.Type == api.SecretTypeServiceAccountToken {
+				glog.V(2).Infof("Using Getting secret named: %s", secret.Name)
```
Comment (font) [inline pkg/kubefed/join.go]: s/Getting //

**Review by font — COMMENTED** [review]
Works on `minikube` with `--federation-config-from-cluster=true`.

/lgtm

File: test/e2e/framework/cluster.go (hunk 1 above)
Comment (marun) [reply test/e2e/framework/cluster.go]: Done

**Comment by shashidharatd** [comment]
@marun, please resolve the conflict. lgtm otherwise. would apply the lgtm as soon the conflict is resolved.

**Pull request merged**